RELEASE.md

```diff
@@ -57,7 +57,7 @@ CI is wired to ["deploy"](https://github.com/Datatamer/unify-client-python/blob/
 
 Check that CI tests passed.
 Check that CI successfully published the release version to [PyPI](https://pypi.org/project/tamr-unify-client/#history).
-
+Add release date to `CHANGELOG.md`.
```

Comment:
I think adding a newline before this will help readability once the markdown gets rendered. Also, let's specify that this should be done on the `master` branch and not the `release` branch.

```suggestion

On the `master` branch, add release date for this release in the `CHANGELOG.md`.
```